Set up SIR development via Docker Compose environment

yvanzo opened the pull request:
## Problem

Previously, running a different version of SIR in development setup was requiring either to modify the `Dockerfile` or to add a local compose file setting build arguments, making it tedious to switch from one version to another.

## Solution

- Prefix all variables for SIR development setup with `SIR_DEV_` so that they don’t need to be unset when switching from SIR local clone to SIR current release.
- Add Docker Compose environment variables to set SIR build arguments in development setup.

### Additionally

Fix setting environment variables in SIR development container for use with its entry point.
Set `../sir` as the default value for the path to the Git clone of `sir` repository.

## Tests

In a setup installed for local development of MusicBrainz Server.

1. Tested checking out the tag `v3.0.1` in a clone of `sir` repository created under the same parent directory as `musicbrainz-docker` clone, and then following the instructions.

   Inspected `indexer` container for variables, followed its logs until _Runit started_, and reindexed `instrument` successfully.

2. Tested checking out the branch `2to3` (currently at commit `b2ed6dbc3b983c92f46adf8fd80e56abecba7a4a`) in the same clone of `sir`, setting the below variables in the file `.env`, and then following the instructions.

   ```env
   SIR_DEV_PYTHON_VERSION=3.11
   SIR_DEV_BASE_IMAGE_DATE=20231006
   SIR_DEV_VERSION=2py3-gb2ed6db
   ```

   Inspected `indexer` container for variables, followed its logs until _Runit started_, and reindexed `instrument` with _Unsupported scheme ''_ error which has to be investigated, more likely in `sir` repository than here.

yvanzo committed:
Amend 1f0218c3: Set variables for sir entrypoint

yvanzo committed:
Set default path to sir clone for development

Previously, `SIR_LOCAL_ROOT` had to be set in the `.env` file,
mandatorily.

The default path is now `../sir` which assumes that `musicbrainz-docker`
and `sir` have been cloned under the same parent directory.

yvanzo committed:
Allow running any version of SIR via configuration

Previously, running a different version of SIR was requiring either to
modify the Dockerfile or to add a local compose file setting build arguments.

This patch allows specifying the version of SIR, the Python version and
the date of its base image, through build arguments and Docker Compose
environment variables for SIR development setup only as needed:

- `SIR_DEV_PYTHON_VERSION` (matching a `metabrainz/python` image tag)
- `SIR_DEV_BASE_IMAGE_DATE` (matching a `metabrainz/python` image tag)
- `SIR_DEV_VERSION` (informative only)

Rebuilding the image is required after changing build arguments.

yvanzo committed:
Prefix all SIR dev setup variables with `SIR_DEV_`

Prefixing variables with `SIR_DEV_` has two purposes:
- It clarifies that it is used for SIR development setup only.
- It allows to keep two separate configuration files for SIR,
  one for the current version, one for the development version,
  thus making it quicker to switch from one setup to the other.

amCap1712 reviewed (commented):
I tested the changes and the Unsupported scheme error is because of the empty sentry dsn here: https://github.com/metabrainz/musicbrainz-docker/blob/831bfe3ba680284d251919f5f6a291e638278fca/default/indexer.ini#L28. We should update SIR to not initialize sentry if dsn is not specified.

amCap1712 commented:
I looked into it a bit more and realised that the DSN is being read literally as `""` by the configparser, not as an empty string as one would expect. I am guessing new versions of sentry are do stricter checking on DSN format. Fix should be simple, replacing the referenced line with `dsn = `

yvanzo committed:
Fix omitting Sentry DSN in SIR config

It resolves the following error when running SIR with Python 3:

    Unsupported scheme ''

From @amCap1712:

> the DSN is being read literally as "" by the configparser,
> not as an empty string as one would expect. I am guessing new
> versions of sentry are do stricter checking on DSN format

I tested it with Python 3.11 and then the still used Python 2.7.

yvanzo commented:
Thanks @amCap1712 for the debugging. See last commit message.
